vignettes/download_jira_issues.Rmd

```diff
@@ -0,0 +1,367 @@
+
+Username: jondoe@jondoe.com
+Password: jondoespassword
+
+Details on [how to obtain an atlassian token](https://support.atlassian.com/atlassian-account/docs/manage-api-tokens-for-your-atlassian-account/)
+
+```{r}
+conf <- yaml::read_yaml("../conf/geronimo.yml")
+# Project domain
+issue_tracker_domain <- conf[["issue_tracker"]][["jira"]][["domain"]]
+# Project key
+issue_tracker_project_key <- conf[["issue_tracker"]][["jira"]][["project_key"]]
+# Altered save paths. Important for naming conventions
+save_path_issue_tracker_issues <- paste0(conf[["issue_tracker"]][["jira"]][["issues"]], issue_tracker_project_key, "_issues")
+save_path_issue_tracker_issue_comments <- paste0(conf[["issue_tracker"]][["jira"]][["issue_comments"]], issue_tracker_project_key, "_issue_comments")
+# Unaltered save paths from config file for use with refresh function
+refresh_issues <- conf[["issue_tracker"]][["jira"]][["issues"]]
+refresh_issue_comments <- conf[["issue_tracker"]][["jira"]][["issue_comments"]]
+# Read the credentials from the file
+# Assuming the first line is the username and the second line is the password
+# Username: jondoe@jondoe.com
+# Password: jondoespassword
+credentials <- scan("~/.ssh/atlassian_credentials", what = "character", quiet = TRUE)
+```
+
+## Create the directories specified in the config files. 
+
+The file schema of files for JIRA projects will be ../../rawdata/JIRA/project_name/issues or ../../rawdata/JIRA/project_name/issues_comments
+
+If you have already created your directories, there is no need to run this chunk. 
+```{r eval = FALSE}
+# Check if the directory for issues exists, create if it doesn't
+if (!dir.exists(refresh_issues)) {
+  dir.create(refresh_issues)
+  message("Directory ", refresh_issues, " created")
+} else {
+  message("Directory ", refresh_issues, " already exists")
+}
+
+# Check if the directory for issue comments exists, create if it doesn't
+if (!dir.exists(refresh_issue_comments)) {
+  dir.create(refresh_issue_comments)
+  message("Directory ", refresh_issue_comments, " created")
+} else {
+  message("Directory ", refresh_issue_comments, " already exists")
+}
+```
+
+## Download issues
+
+This calls a function that downloads all issue data (without comments) from a repository specified in the config file. 
+
+max_result is the number of issues downloaded per page, verbose when set to TRUE prints operational messages, and max_downloads sets an upper limit on how many issues will be downloaded within an hour before the function ceases API calls. You can change these values or keep them the same for simplicity. If there are more issues than the value of max_downloads, run the issue refresh function to continue downloading.
```

Comment:
We should mention here where to obtain the list of fields the endpoint accepts should the user want to know what there is out there, and note the parser will not handle currently.

Reply:
https://developer.atlassian.com/cloud/jira/platform/rest/v2/api-group-issue-fields/#api-group-issue-fields This endpoint can return a list of fields for a given issue. I wonder if there is a glossary somewhere?